src/transport.ts

```diff
@@ -74,25 +80,32 @@ export class WebRTCTransport implements Transport, Initializable {
     //
     // construct answer sdp from multiaddr
     let answerSdp = sdp.fromMultiAddr(ma, ufrag);
-    //
+    
     //
     //
     // set remote description
     peerConnection.setRemoteDescription(answerSdp);
+
     //
     //
     //
     // wait for peerconnection.onopen to fire, or for the datachannel to open
     let dataChannelOpenPromise = defer();
+
     handshakeDataChannel.onopen = (_) => dataChannelOpenPromise.resolve();
-    setTimeout(dataChannelOpenPromise.reject, 10000);
+    handshakeDataChannel.onerror = (ev: Event) => {
+      log.error('Error opening a data channel for handshaking: %s', ev.toString()); 
+        dataChannelOpenPromise.reject();
+      };
+    setTimeout(() => {
+        log.error('Data channel never opened. State was: %s', handshakeDataChannel.readyState.toString()); 
+        dataChannelOpenPromise.reject() 
+      }, 10000);
```

Comment:
Possible magic number.  Can you assign it to a variable?  Also, can you comment on the significance of the number chose?

Reply:
It's a completely arbitrary number. I believe Chinmay used it when he was demonstrating the technique, and I stole it. 

10 seconds does work as a timeout in that it's considerably longer than this process should ever take. But also, yes, it would be completely reasonable to name this variable.